js/quotechew.js

```diff
@@ -113,8 +130,42 @@ var RE_SECTION_DELIM = /^[_-]{6,}$/;
 
 var RE_LIST_BOILER = /mailing list$/;
 
+/**
+ * Non-localized hard-coded heuristic to help us detect whether a line is the
+ * lead-in to a quote block.  This is only checked against the last content line
+ * preceding a quote block.  This only results in styling guidance, so if it's
+ * too aggresive or insufficiently aggressive, the worst that happens is the
+ * line will be dark gray instead of black.  (For now!  Moohoohahahahahaha.)
+ */
 var RE_WROTE_LINE = /wrote/;
 
+/**
+ * Hacky reply helper to check if the last line in a content block contains the
+ * words wrote.  In a nutshell:
+ * - The current quoting logic can only detect a quoting lead-in in non-quoted
+ *   text and I'm not crazy enough to enhance this at this time.  (The current
+ *   mechanism is already complex and assumes it's not dealing with '>'
+ *   characters.  An enhanced mechanism could operate on a normalized
+ *   representation where the quoting has been normalized to no longer include
+ *   the > characters.  That would be a sortof multi-pass AST/tree-transformer.
+ *   Furture work.)
+ * - The reply logic performs normalization on output, but without knowing about
+ *   lead-ins, it makes the wrong normalization call for these wrote lines,
+ *   inserting wasted whitespace.  This regex is an attempt to retroactively do
+ *   the cool thing the former bullet proposes.  And it does indeed work out
+ *   because of our aggressive newline normalization logic.
+ * - Obviously this is not localized at all, but neither is the above.
+ *
+ * To keep the regex cost-effective and doing what we want, we anchor it to the
+ * end of the string and ensure there are no newline characters before wrote.
+ * (And we check for wrote without any other constraints because that's what the
+ * above does.)
+ *
+ * Again, since this only controls whether a newline is inserted or not, it's
+ * really note the end of the world.
+ */
+var RE_REPLY_LAST_LINE_IN_BLOCK_CONTAINS_WROTE = /wrote[^\n]+$/;
```

Comment:
For these "wrote" regexps, is it worth including the colon in the regexp, so that "wrote:" is the match vs a loose wrote in the line? As the comment says, this is not the end of the world on a false positive, so does not need to change, mostly inquiring to know how much variability you know of happens for those "wrote" lines, if colons are not guaranteed.